### PR TITLE
docs: better README and CONTRIBUTING

### PR DESCRIPTION
**Pull request by jackdelahunt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,227 @@
...
+Python packages and API clients for Red Had OpenShift Application Services (RHOAS) 
+
+## Prequisites
+
+- [Python 3.9](https://docs.python.org/3/) or above
+- [pip](https://pypi.org/project/pip/) for installing packages
+
+## Installation
+
+Currently all RHOAS SDKs are bundled together. To install the RHOAS SDK with the pip package installer:
```
Comment (andreaTP) [inline README.md:75]: this is actually surprising, should we have a ticket to improve here? Is Python the only exception?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,227 @@
...
+## Examples
+[Examples](/app-services-sdk-python/examples)
+
+# RHOAS SDK for Java
+
+Java packages and API clients for RHOAS services
+
+## Prequisites
+
+- [NodeJS 14.x lts](https://nodejs.org/en/about/releases/) or above
```
Comment (andreaTP) [inline README.md:113]: Why do we require Node to build Java?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,227 @@
...
+- [Maven 3.6.x](https://maven.apache.org) or above
+- [YQ 4.6.x](https://github.com/mikefarah/yq#install) or above
```
Comment (andreaTP) [inline README.md:115]: Why?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,227 @@
...
+[Examples](/app-services-sdk-python/examples)
+
+# RHOAS SDK for Java
+
+Java packages and API clients for RHOAS services
+
+## Prequisites
+
+- [NodeJS 14.x lts](https://nodejs.org/en/about/releases/) or above
+- [Maven 3.6.x](https://maven.apache.org) or above
```
Comment (andreaTP) [inline README.md:114]: I think we are missing a Java JDK distribution mention.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,227 @@
...
+
+# RHOAS SDK for Go
+
+[![Go Reference](https://pkg.go.dev/badge/github.com/redhat-developer/app-services-sdk-go.svg)](https://pkg.go.dev/github.com/redhat-developer/app-services-sdk-go)
+
+Go packages and API clients for Red Hat OpenShift Application Services (RHOAS) 
+
+## Prequisites
+
+- [Go 1.15](https://golang.org/doc/go1.15) or above
```
Comment (andreaTP) [inline README.md:151]: `1.15` ?

File: app-services-sdk-python/sdks/smart_events_mgmt_sdk/README.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+ - **openid**: Treat as an OIDC request
+ - **api.iam.service_accounts**: Grants access to the service accounts api
+
+
+## Author
+
+it-user-team-list@redhat.com
+
+
+## Notes for Large OpenAPI documents
```
Comment (andreaTP) [inline app-services-sdk-python/sdks/smart_events_mgmt_sdk/README.md:142]: What?

**Review by andreaTP — COMMENTED** [review]
Thanks for going through this!
Left a few initial comments, but, feel free to merge and improve on top if needed!

File: README.md (hunk 5 above)
Comment (jackdelahunt) [reply README.md:151]: The go SDK in core uses `go 1.15` in the mod file so to use it you need to at least be able to compile go `1.15` with what ever version of the compiler you have. Is this wrong maybe I am misunderstanding how it works

File: README.md (hunk 1 above)
Comment (jackdelahunt) [reply README.md:75]: Yes currently python is the only one that combines them all, I don't think there was any plans to change this @dimakis when you created the python sdk was there a specific reason for this difference?

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,5 @@
+generate:
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply Makefile:1]: I didn't know the auto-generated help existed nice 👍

[the remark this answers is not in the document]
Comment (jackdelahunt) [reply Makefile:1]: What would the `make clean` do? I don't think we have unneeded files from generating SDKs. Unless if you mean to remove the current SDKs maybe?

[the remark this answers is not in the document]
Comment (andreaTP) [reply Makefile:1]: what happens if an api removes an endpoint?
Just asking 🙂

File: README.md (hunk 5 above)
Comment (andreaTP) [reply README.md:151]: fair enough 👍

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -20,6 +12,28 @@ When releasing service to production we need to trigger documentation portal rel
...
+## Generating API clients locally
+
+```bash
+make generate
+```
+
+## Checking standards
+
+```bash
+spectral lint ./.openapi/kafka-admin-rest.yaml
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply CONTRIBUTING.md:34]: Done

File: README.md (hunk 2 above)
Comment (jackdelahunt) [reply README.md:113]: We use `npx` to run the openapi generator

File: README.md (hunk 3 above)
Comment (jackdelahunt) [reply README.md:115]: These are requirements for developing I will move these to `CONTRIBUTING`

File: README.md [hunk 9]
```diff
@@ -1,6 +1,227 @@
...
+Typically developers need to integrate with Management SDK's to create or manage service instances 
+and later use upstream libraries or management SDK's to interact with services.
+
+You can view all SDK packages directly in [npm](https://www.npmjs.com/search?q=keywords:rhoas)
+
+## Management SDK's
+
+| Name             | Package name                        | Status | Version            | NPM                 | Docs                |
+| ---------------- | :---------------------------------- | :----- | ------------------ | ------------------- | ------------------- |
+| kafka            | **@rhoas/kafka-management-sdk**     | beta   | ![kafkashield]     | [npm][kafkanpm]     | [doc][kafkagit]     |
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply README.md:42]: I am not sure how we decide on the current status of each SDK. These were carried over from the original SDKs. It seems like all instance SDKs and the kafkamgmt SDK are in beta for each language and the reast are in alpha. @dimakis might have more info on this

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -20,6 +12,28 @@ When releasing service to production we need to trigger documentation portal rel
...
+
+## Checking standards
+
+```bash
+spectral lint ./.openapi/kafka-admin-rest.yaml 
+```
+
+For more info see:
+
+<http://api.appservices.tech/spectral/>
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply CONTRIBUTING.md:47]: Removed as now I just ask to install the rhoas wrapper for spctral

File: README.md (hunk 4 above)
Comment (jackdelahunt) [reply README.md:114]: Can you expand on this further? Do you mean requirements for publishing or just using in a project?

Comment (andreaTP) [reply README.md:114]: Having a JDK is a requirement to build, as we install it in the CI:
https://github.com/redhat-developer/app-services-sdk-core/blob/91fb90ecdd8c90fa18b95fa03f780e60d7e0571b/.github/workflows/release-java.yaml#L16

File: README.md (hunk 2 above)
Comment (andreaTP) [reply README.md:113]: should it be a Prerequisite for all of the langs?

File: README.md (hunk 1 above)
Comment (andreaTP) [reply README.md:75]: discussed and agreed to consolidate those decisions while working on Kiota

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (andreaTP) [reply README.md:42]: Can you open a separate issue to review those sections?

**Review by andreaTP — APPROVED** [review]
Please @jackdelahunt capture the comments that are not addressed in separate issues and let's move on with this!
Thanks for the improvement!

File: README.md [hunk 11]
```diff
@@ -1,6 +1,227 @@
...
-underlying RHOAS SDKS (<https://github.com/topics/rhoas-sdk>)
+underlying RHOAS SDKS. 
+
+# RHOAS SDK for Typescript and JavaScript
+
+Typescript and JavaScript packages and API clients for RHOAS services
+
+## Prequisites
+
+- [NodeJS 14.x lts](https://nodejs.org/en/about/releases/) or above
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply README.md:31]: https://github.com/redhat-developer/app-services-sdk-core/issues/117

File: README.md [hunk 12]
```diff
@@ -1,6 +1,227 @@
...
+- **Management SDKs** - support creating services
+- **Instance SDKs** -  support interaction with services (limited to capabilities not supported by upstream tools)
+
+## Purpose of this repository
 
 Repository contains source code for openapi based generator along with automation scripts that generate all
-underlying RHOAS SDKS (<https://github.com/topics/rhoas-sdk>)
+underlying RHOAS SDKS. 
+
+# RHOAS SDK for Typescript and JavaScript
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply README.md:25]: https://github.com/redhat-developer/app-services-sdk-core/issues/118

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply README.md:37]: https://github.com/redhat-developer/app-services-sdk-core/issues/119

File: README.md [hunk 13]
```diff
@@ -1,6 +1,227 @@
...
+Typescript and JavaScript packages and API clients for RHOAS services
+
+## Prequisites
+
+- [NodeJS 14.x lts](https://nodejs.org/en/about/releases/) or above
+
+Typically developers need to integrate with Management SDK's to create or manage service instances 
+and later use upstream libraries or management SDK's to interact with services.
+
+You can view all SDK packages directly in [npm](https://www.npmjs.com/search?q=keywords:rhoas)
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply README.md]: Done

**Pull request merged**